KSystemInformer/dynimp.c

```diff
@@ -49,6 +50,46 @@ VOID KphDynamicImport(
     KphDynCiValidateFileObject = (PCI_VALIDATE_FILE_OBJECT)KphGetRoutineAddress(L"ci.dll", "CiValidateFileObject");
     KphDynCiFreePolicyInfo = (PCI_FREE_POLICY_INFO)KphGetRoutineAddress(L"ci.dll", "CiFreePolicyInfo");
     KphDynLxpThreadGetCurrent = (PLXP_THREAD_GET_CURRENT)KphGetRoutineAddress(L"lxcore.sys", "LxpThreadGetCurrent");
+
+#ifdef _WIN64
+    __try
+    {
+        // ObGetObjectType have equal machine code on 10.0.10240 - 10.0.22631
+        //
+        // nt!ObGetObjectType+0x1C      488d0d[????????]   lea     rcx,[nt!ObTypeIndexTable]
+        // lea  rcx, [rip + off32]
+        PUCHAR rip;
+
+        rip = (PUCHAR)ObGetObjectType + 0x1C;
```

Comment:
It was experimental implementation, and I know what it is bad and not safe. I wrote this with a purpose to open objects for which public `NtOpenXXX `API isn't existing (like **Type**, Callback). This gives a power to open any object.
Object types table offset is RIP relative to instruction in `ObGetObjectType`. To add it to dyndata we need to know `ObGetObjectType` address, `lea  rcx` instruction address and offset vaule, then we can calculate full address. And we should to it at runtime, since it address points to data section of loaded kernel image (hope I'm not wrong).

It's very interesting, but for now there is no particular need of `KphOpenObjectByTypeIndex`. Type information and statistics can be retrieved without need to open this Type object. Current implementation always open object properties, even if object cannot be opened.

But, if possible, I rally want to keep `PhOpenDevice` (see answer below).

Example, `\ObjectTypes\Token` information with driver and KphOpenObjectByTypeIndex (left), and without (right):

https://github.com/user-attachments/assets/1214bfb4-de3e-4079-a335-7c072b987f47